evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -1881,3 +1882,48 @@ def test_component_graph_with_X_y_inputs_y(mock_fit, mock_fit_transform):
     assert_series_equal(mock_fit_transform.call_args[0][1], y)
     # Check that we use "Log.y" for RF
     assert_series_equal(mock_fit.call_args[0][1], infer_feature_types(np.log(y)))
+
+
+@pytest.mark.parametrize(
+    "pipeline_parameters,set_values",
+    [
+        ({"Logistic Regression Classifier": {"penalty": "l1"}}, {}),
+        ({"Logistic Regression": {"penalty": "l1"}}, {"Logistic Regression"}),
```

Comment:
This should issue a warning since the component name we've defined in the graph is `Logistic Regression Classifier` rather than `Logistic Regression`